guide/samples/tests/storage/rewrite_object.rs

```diff
@@ -0,0 +1,132 @@
+// Copyright 2025 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// ANCHOR: all
+use gcs::Result;
+use gcs::builder::storage_control::RewriteObject;
+use gcs::client::StorageControl;
+use gcs::model::Object;
+use google_cloud_gax::retry_policy::AlwaysRetry;
+use google_cloud_storage as gcs;
+
+pub async fn rewrite_object(bucket_name: &str) -> anyhow::Result<()> {
+    let source_object = upload(bucket_name).await?;
+
+    // ANCHOR: client
+    let control = StorageControl::builder()
+        .with_retry_policy(AlwaysRetry)
+        .build()
+        .await?;
+    // ANCHOR_END: client
+
+    // ANCHOR: builder
+    let mut builder = control
+        .rewrite_object()
+        .set_source_bucket(bucket_name)
+        .set_source_object(&source_object.name)
+        .set_destination_bucket(bucket_name)
+        .set_destination_name("rewrite-object-clone");
+    // ANCHOR_END: builder
+
+    // Optionally limit the max bytes written per request.
+    // ANCHOR: limit-bytes-per-call
+    builder = builder.set_max_bytes_rewritten_per_call(1024 * 1024);
+    // ANCHOR_END: limit-bytes-per-call
+
+    // Optionally change the storage class to force GCS to copy bytes
+    // ANCHOR: change-storage-class
+    builder = builder.set_destination(Object::new().set_storage_class("NEARLINE"));
```

Comment:
This example will cost 30 days of storage:

https://cloud.google.com/storage/docs/storage-classes#classes
https://cloud.google.com/storage/pricing-examples#early-delete

It is way below $0.01[^1], but we should warn people "If you run this example with larger data consider the costs [link]".

[^1]: https://cloud.google.com/products/calculator?hl=en&dl=CjhDaVJtWW1NNU5XTTRZaTFrTVRnNExUUXdObVl0WVRNME1DMDROelUzWmpBelpqRXlOemNRQVE9PRAJGiRFNDc2MTMyMy04NzBBLTRGRUYtOTkyOS05OUFDNTE2QjlEMEQ